alf/utils/data_buffer.py

```diff
@@ -245,22 +246,38 @@ def _enqueue(self, batch, env_ids=None):
                 "batch and env_ids do not have same length %s vs. %s" %
                 (batch_size, env_ids.shape[0]))
 
-            # Make sure that there is no duplicate in `env_id`
-            # torch.unique(env_ids, return_counts=True)[1] is the counts for each unique item
-            assert torch.unique(
-                env_ids, return_counts=True)[1].max() == 1, (
-                    "There are duplicated ids in env_ids %s" % env_ids)
+            current_pos = self._current_pos[0].item()
+            if (env_ids.numel() == self._num_envs
+                    and (self._current_pos == current_pos).all()
+                    and (env_ids == self._env_ids).all()):
+                # fast path for the common case
```

Comment:
explain a bit more in the comment, that this common case is when all envs have the same current_pos, and when all envs unroll the same number of steps.